search-java/src/main/java/uk/os/elements/search/android/providers/addresses/Transform.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2016 Ordnance Survey
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.os.elements.search.android.providers.addresses;
+
+import com.esri.core.geometry.Envelope;
+import com.esri.core.geometry.Point;
+import com.esri.core.geometry.SpatialReference;
+import rx.Observable;
+import rx.functions.Func1;
+import rx.functions.FuncN;
+import uk.os.elements.search.SearchResult;
+import uk.os.elements.search.android.providers.addresses.service.model.Result;
+import uk.os.elements.search.android.providers.addresses.service.model.ServerResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+final class Transform {
+
+    protected static Observable<List<SearchResult>> toSearchResults(List<Observable<ServerResponse>> results) {
+
+        // Transform list of ServerResponses to a list of SearchResults
+        List<Observable<List<SearchResult>>> transformedResults = new ArrayList<>();
+        for (Observable<ServerResponse> result : results) {
+            Observable<List<SearchResult>> transformed = result.map(searchResultsFromServerResponse());
+            transformedResults.add(transformed);
+        }
+
+        // Zip SearchResult lists
+        Observable<List<SearchResult>> zipped = Observable.zip(transformedResults, new FuncN<List<SearchResult>>() {
+
+            @SuppressWarnings("unchecked")
```

Comment:
"The FuncN type is used in places...where heterogeneous types are supported and there are more than 9 types (supported by Func1 through Func9). These are combineLatest and zip, both of which have strongly typed signatures up to 9 types."   Source: https://github.com/ReactiveX/RxJava/issues/1209.

The problem in this case is 'transformedResults' is a list of N streams, which we are zipping (think merging) into one.  So if we had 1-9 streams then we could avoid the warning.

otr, I am v.tempted to change this to kotlin and simplify.  Unlike some of the providers, this hasn't had much love.